### ISSUE TITLE
Multi-GPU support

### ISSUE THREAD

**Issue by beccamc** [dotnet/machinelearning#6453]
Reported by @torronen in tools repo - [original issue](https://github.com/dotnet/machinelearning-modelbuilder/issues/2364). Coping the request for Multi-GPU support here. 

New Text Classification can efficiently use GPU, almost 100% according to task manager. Need for CPU is also quite high, so it probably would be the next bottleneck, but multi-gpu support would be nice to lower temperatures and at least some speed up.

![image](https://user-images.githubusercontent.com/26261427/201221706-df2c20ab-1d7f-4efe-893f-7a4036142f7a.png)

**Comment by torronen** [comment]
Text Classification seems slow, so far 2-3 weeks running. Some Keras models were trained on this dataset in 1-2 days before. For now, I am using Model Builder and I did not yet try with the class library, but maybe there are hyperparameters to make it faster. Anyway, setting up a multi-gpu at Azure or another vendor seems to be enticing. So, I suppose the more useful use case could be 8 GPU's in a cloud environment and not 2 GPUs in a desktop.

**Comment by michaelgsharp** [comment]
@torronen how big of a dataset are you using? Are you in ML.NET itself or in model builder or automl? I'm trying to see where the bottleneck is. Do you know if PyTorch/TorchSharp supports multi-gpu? I would bet PyTorch does, not sure about TorchSharp though.

**Comment by torronen** [comment]
@michaelgsharp 
I am using Model Builder, and I do not have experience / knowledge about TorchSharp yet. I also have not yet tried with ML.NET.

3 557 109 rows. Column "NormalizedDescription" is used to predict one out of 10 categories. It is a text column with 1-2 paragraphs of text. In this case Normalized means the baseword / lemma of the words. CSV file size is about 1.1 gigabytes. There are 9 other columns but they are ignored in the Model Builder settings. Actually, I could not get any of the more specific classifications to work (I think, it is about 2 char = 100 classes, 3 char = 500, 4 char =3500)

The task is to classify job ads according to ESCO classification.



![image](https://user-images.githubusercontent.com/26261427/204594253-f1a585b5-d4a1-4750-af6b-e030c5ca5263.png)
